### ISSUE TITLE
Normalizer can't handle license files properly

### ISSUE THREAD

**Issue by guenhter** [jk1/Gradle-License-Report#52]
There are a few renderers (CSV, JSON, XML) which rely on the following logic:

```
String moduleLicense = null
String moduleLicenseUrl = null
def text = new File(config.outputDir, it).text
if (text.contains('Apache License, Version 2.0')) {
    moduleLicense = 'Apache License, Version 2.0'
    moduleLicenseUrl = 'http://www.apache.org/licenses/LICENSE-2.0'
}
if (text.contains('Apache Software License, Version 1.1')) {
    moduleLicense = 'Apache Software License, Version 1.1'
    moduleLicenseUrl = 'http://www.apache.org/licenses/LICENSE-1.1'
}
if (text.contains('CDDL')) {
    moduleLicense = 'COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (CDDL) Version 1.0'
    moduleLicenseUrl = 'http://opensource.org/licenses/CDDL-1.0'
}

if (moduleLicense || moduleLicenseUrl) {
    info.licenses << new License(name: moduleLicense, url: moduleLicenseUrl)
}
```

This makes a problem for the normalisation, because this is after normalizing. 
One way to solve this, is to extend the current `LicenseFileData` about the fields `moduleLicense` and `moduleLicenseUrl`, and add support in the normalisation to define file-content-patterns. When a pattern machtes, the two new fields are then filled, and everything is fine.
For backward compatibility reason, the code from above stays, but is extedned with a check, so that it is only executed when the two new fields are empty.
@jk1 what do you say to that?

**Comment by jk1** [comment]
While the first option looks good to me I would also consider the following approach: 
Extract this code and move it to the parsing part to make it execute before the normalization occurs.

Backward compatibility doesn't seem to be a big deal in this particular case.

**Comment by guenhter** [comment]
Solved by #59

**Comment by jk1** [comment]
#59 has been accepted and merged
